pkg/agent/nodeportlocal/k8s/controller.go

```diff
@@ -112,12 +173,122 @@ func (c *Controller) EnqueueObj(obj interface{}) {
 	c.queue.Add(podKey)
 }
 
-func (c *Controller) Worker() {
+func (c *NPLController) checkDeletedSvc(obj interface{}) (*corev1.Service, error) {
+	deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+	if !ok {
+		return nil, fmt.Errorf("Received unexpected object: %v", obj)
+	}
+	svc, ok := deletedState.Obj.(*corev1.Service)
+	if !ok {
+		return nil, fmt.Errorf("DeletedFinalStateUnknown object is not of type Service: %v", deletedState.Obj)
+	}
+	return svc, nil
+}
+
+func (c *NPLController) enqueueSvcUpdate(oldObj, newObj interface{}) {
+	// In case where the app selector in Service gets updated from one valid selector to another
+	// both sets of Pods (corresponding to old and new selector) need to be considered
+
+	// Donot push to queue if 1) Service ResourceVersions don't change OR
+	// 2) Service spec selectors AND 3) NPLEnabledAnnotationKey values don't change
```

Comment:
Probably we should remove "3)", as the two conditions together define "2)"?

Reply:
Updates to .spec.selector and .metadata.annotations can happen separately.